libs/base/Decidable/Equality.idr

```diff
@@ -111,6 +111,17 @@ public export
 DecEq a => DecEq (List1 a) where
   decEq (x ::: xs) (y ::: ys) = decEqCong2 (decEq x y) (decEq xs ys)
 
+--------------------------------------------------------------------------------
+-- SnocList
+--------------------------------------------------------------------------------
+
+public export
+DecEq a => DecEq (SnocList a) where
+  decEq Lin Lin = Yes Refl
+  decEq (xs :< x) Lin = No absurd
+  decEq [] (xs :< x) = No absurd
```

Comment:
Did you want `[<]` instead of `[]` here? Types should not match with this

Reply:
yes, fixed. Is `[<]` the same as `Lin`?

Reply:
> yes, fixed. Is `[<]` the same as `Lin`?

It is